### ISSUE TITLE
About the use of MixConv2d

### ISSUE THREAD

**Issue by Zengyf-CVer** [ultralytics/yolov5#5403]
@glenn-jocher 
How should I use this MixConv2d method? I did an experiment and replaced Conv directly with the MixConv2d method, and found that it couldn't run:
```yaml
backbone:
  # [from, number, module, args]
  [[-1, 1, Conv, [64, 6, 2, 2]],  # 0-P1/2
   [-1, 1, MixConv2d, [128, 3, 2]],  # 1-P2/4
   [-1, 3, C3, [128]],
   [-1, 1, MixConv2d, [256, 3, 2]],  # 3-P3/8
   [-1, 6, C3, [256]],
   [-1, 1, MixConv2d, [512, 3, 2]],  # 5-P4/16
   [-1, 9, C3, [512]],
   [-1, 1, MixConv2d, [1024, 3, 2]],  # 7-P5/32
   [-1, 3, C3, [1024]],
   [-1, 1, SPPF, [1024, 5]],  # 9
  ]
```
Console output information:
![ksnip_20211030-000732](https://user-images.githubusercontent.com/41098760/139467911-7bb57ab3-9a54-4c71-8bd5-c1fd8a63003a.png)

I found that the problem with this `k`, how do I set this `k`?
https://github.com/ultralytics/yolov5/blob/ed887b5976d94dc61fa3f7e8e07170623dc7d6ee/models/experimental.py#L53

**Comment by glenn-jocher** [comment]
@Zengyf-CVer see MixConv2d module for input requirements:
https://github.com/ultralytics/yolov5/blob/ed887b5976d94dc61fa3f7e8e07170623dc7d6ee/models/experimental.py#L49-L71

**Comment by glenn-jocher** [comment]
@Zengyf-CVer Conv() module for comparison:
https://github.com/ultralytics/yolov5/blob/ed887b5976d94dc61fa3f7e8e07170623dc7d6ee/models/common.py#L36-L46

**Comment by Zengyf-CVer** [comment]
@glenn-jocher 
I used a closed test and found a problem. Is this a bug?
```python
import torch
from utils.torch_utils import profile
from models.experimental import MixConv2d

m = MixConv2d(128, 256, (3, 5), 1)
results = profile(input=torch.randn(16, 128, 80, 80), ops=[m], n=1)
```
Error message:
```
The size of tensor a (128) must match the size of tensor b (256) at non-singleton dimension 1
```
Is there a bug for MixConv2d?

**Comment by glenn-jocher** [comment]
@Zengyf-CVer I cam able to reproduce this error (thank you for the code to reproduce!), may be a bug, will investigate.

**Comment by glenn-jocher** [comment]
TODO: Possible MixConv2d() bug

**Comment by glenn-jocher** [comment]
@Zengyf-CVer based on a re-read of the MixConv paper I don't understand why I put a shortcut into the current module. The reason for the error is that the forward function includes a sum of the input to the output, and the input and outputs of course have different channel counts (128 and 256), causing the error. I will remove the shortcut and push a fix.
```pyhon
    def forward(self, x):
        return x + self.act(self.bn(torch.cat([m(x) for m in self.m], 1)))
```

<img width="1876" alt="Screenshot 2021-10-30 at 13 01 46" src="https://user-images.githubusercontent.com/26833433/139530273-7a1a19f3-157a-4e3b-a8cc-931716f43114.png">

<img width="1965" alt="Screenshot 2021-10-30 at 13 01 09" src="https://user-images.githubusercontent.com/26833433/139530257-b90fb04c-6cfd-416e-b1f1-fa7f63fa1116.png">

**Comment by glenn-jocher** [comment]
@Zengyf-CVer good news 😃! Your original issue may now be fixed ✅ in PR #5410. To verify this fix:

```python
import torch
from utils.torch_utils import profile
from models.experimental import MixConv2d
from models.common import Conv

m1 = MixConv2d(128, 256, (3, 5), 1)
m2 = Conv(128, 256, 3, 1)
results = profile(input=torch.randn(16, 128, 80, 80), ops=[m1, m2], n=3)

YOLOv5 🚀 v6.0-39-g3d9a368 torch 1.9.0+cu111 CUDA:0 (Tesla P100-PCIE-16GB, 16280.875MB)

      Params      GFLOPs  GPU_mem (GB)  forward (ms) backward (ms)                   input                  output
        4864      0.9961         0.684         4.922         12.76       (16, 128, 80, 80)       (16, 256, 80, 80)
      295424        60.5         0.990         9.727         8.917       (16, 128, 80, 80)       (16, 256, 80, 80)
```

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
